### ISSUE TITLE
ocamlfind: Hunk #1 FAILED on findlib 1.9.5

### ISSUE THREAD

**Issue by ralsei** [balsoft/opam-nix#11]
**Describe the bug**
When trying to build `ocamlfind` as part of an `opam-nix` derivation, the following error arises:
```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/jdhi5amjwlp8ymf0c7f342ax42b6f8s4-findlib-1.9.5.tar.gz
source root is findlib-1.9.5
setting SOURCE_DATE_EPOCH to timestamp 1655063092 of file findlib-1.9.5/src/findlib/topfind_rd1.p
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
applying patch /nix/store/xcdgajkwqw7jfqyll1pqjf53610nqafs-ldconf.patch
patching file findlib.conf.in
applying patch /nix/store/xs144bvl5yy4530n1aa4dn0grg0jjj3l-install_topfind.patch
patching file src/findlib/Makefile
Hunk #1 FAILED at 123.
1 out of 1 hunk FAILED -- saving rejects to file src/findlib/Makefile.rej
```

I'm not entirely sure where these patches are coming from. Running `nix shell nixpkgs#ocamlPackages.findlib` works fine and puts `ocamlfind` in PATH, so presumably it has something to do with an issue here?

The issue is probably to do with the fact that I'm pulling findlib `1.9.5`, not `1.9.3`. The reason I have a different `opam-repository` as an input is because `bwd` isn't in the outdated version in this repo's `flake.lock`.

**To Reproduce**
Here's my `flake.nix`:
```nix
{
  description = "Experimental implementation of Cartesian cubical type theory";

  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
    flake-utils.url = "github:numtide/flake-utils";

    opam-repository = {
      url = "github:ocaml/opam-repository";
      flake = false;
    };
    
    opam-nix = {
      url = "github:tweag/opam-nix";
      inputs.nixpkgs.follows = "nixpkgs";
      inputs.flake-utils.follows = "flake-utils";
      inputs.opam-repository.follows = "opam-repository";
    };

    kado = {
      url = "github:RedPRL/kado";
      flake = false;
    };
    bantorra = {
      url = "github:RedPRL/bantorra/0.1.0";
      flake = false;
    };
  };

  outputs = { self, flake-utils, opam-nix, nixpkgs, kado, bantorra, opam-repository }@inputs:
    flake-utils.lib.eachDefaultSystem (system: {
      legacyPackages = let
        pkgs = nixpkgs.legacyPackages.${system};
        on = opam-nix.lib.${system};
        scope = on.buildDuneProject {
          repos = [
            (on.makeOpamRepo kado)
            (on.makeOpamRepo bantorra)
            opam-nix.inputs.opam-repository
          ];
        } "cooltt" ./. { 
          ocaml-base-compiler = null;
        };
        overlay = self: super:
          {
            # Your overrides go here
          };
      in scope.overrideScope' overlay;

      defaultPackage = self.legacyPackages.${system}."cooltt";
    });
}
```
and `cooltt.opam`:
```
opam-version: "2.0"
name: "cooltt"
version: "0.0"
maintainer: "Jonathan Sterling <jmsterli@cs.cmu.edu>"
authors: ["The RedPRL Development Team"]
homepage: "https://github.com/RedPRL/cooltt"
bug-reports: "https://github.com/RedPRL/cooltt/issues"
dev-repo: "git+https://github.com/RedPRL/cooltt.git"
synopsis: "Experimental implementation of Cartesian cubical type theory"
license: "Apache-2.0"
depends: [
  "dune" {>= "2.0"}
  "ocaml" {>= "4.10.0"}
  "ppx_deriving" {>= "4.4.1"}
  "bantorra"
  "bwd" {>= "1.2"}
  "cmdliner" {>= "1.1"}
  "containers" {>= "3.4"}
  "ezjsonm" {>= "1.2.0"}
  "menhir" {>= "20180703"}
  "uuseg" {>= "12.0.0"}
  "uutf" {>= "1.0.2"}
  "yuujinchou" {>= "2.0.0" & < "3"}
  "odoc" {with-doc}
  "kado"
]
pin-depends: [
  [ "kado.~dev" "git+https://github.com/RedPRL/kado" ]
  [ "bantorra.0.1.0" "git+https://github.com/RedPRL/bantorra#1e78633d9a2ef7104552a24585bb8bea36d4117b" ]
]
build: [
  ["dune" "build" "-p" name "-j" jobs]
  ["dune" "build" "-p" name "-j" jobs "@runtest"] {with-test}
  ["dune" "build" "-p" name "-j" jobs "@doc"] {with-doc}
]
```
The rest of the code for the actual project is as in https://github.com/RedPRL/cooltt.

**Expected behavior**
ocamlfind should build and install successfully.

**Environment**
 - OS name + version: NixOS 22.05
 - Version of the code: 23c55a8fc0b0448ace7c8be4e1552358a3ffa14c

**Additional context**

**Comment by balsoft** [comment]
Yep, this can happen sometimes. This is because of this override here: https://github.com/tweag/opam-nix/blob/main/src/overlays/ocaml.nix#L49

I'll fix this :)

**Comment by balsoft** [comment]
Should be fixed with 6c8d4581adb9adfda9b2af08e740ff0181114a6a
